lib/blockscore/response.rb

```diff
@@ -16,13 +16,13 @@ def handle_response(resource, response)
     def api_error(response)
       case response.code
       when 400
-        fail InvalidRequestError.new(response)
+        raise InvalidRequestError.new(response)
```

Comment:
Instead of duplicating the same code 4 times I would probably write this as:

```ruby
def api_error(response)
  error_class = case response.code
  when 400 then InvalidRequestError
  when 401 then AuthenticationError
  when 404 then NotFoundError
  else
    APIError
  end
  raise error_class, response
end
```

Note that if you use `raise` with 2 parameters, you don't have to construct an object.

Also, given that this case statement is a simple lookup table, this could be refactored further into a Hash, eg:

```ruby
HTTP_ERRORS = Hash.new(APIError).update(
  401 => AuthenticationError,
  404 => NotFoundError,
)

# ....

def api_error(response)
  HTTP_ERRORS[response.code]
end
```